### PR TITLE
htlcswitch: make mailbox HTLC delivery timeout configurable

### PR DESCRIPTION
**Pull request by yyforyongyu**
While debugging the itest case `async_bidirectional_payments`, I've seen quite some failed attempts due to the mailbox delivery deadline being reached. Thus we'd need to make this value configurable. This probably also happens in real life due to the timeout reached.

Also since we already have the `pending-commit-interval` setting, it seems like this timeout is unnecessary?

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -641,6 +645,9 @@ func DefaultConfig() Config {
 		Sweeper: &lncfg.Sweeper{
 			BatchWindowDuration: sweep.DefaultBatchWindowDuration,
 		},
+		Htlcswitch: &lncfg.Htlcswitch{
```
Comment (Crypt-iQ) [inline config.go:648]: needs to be added to the `lncfg.Validate` call in `ValidateConfig`

Comment (yyforyongyu) [reply config.go:650]: Woops fixed

**Review by Crypt-iQ — APPROVED** [review]
unrelated panic flake?
https://github.com/lightningnetwork/lnd/actions/runs/3320319766/jobs/5486560530

File: config.go [hunk 2]
```diff
@@ -462,6 +464,8 @@ type Config struct {
 }
 
 // DefaultConfig returns all default values for the Config struct.
+//
+// nolint:lll
```
Comment (positiveblue) [inline config.go:468]: Does this mean that the `nolint` applies to all the go function?

if so, this is super cool :tada:

File: lncfg/htlcswitch.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+)
+
+// nolint:lll
+type Htlcswitch struct {
+	MailboxDeliveryTimeout time.Duration `long:"mailboxdeliverytimeout" description:"The timeout value when delivering HTLCs to a channel link. Setting this value too small will result in local payment failures if large number of payments are sent over a short period."`
+}
+
+// Validate checks the values configured for htlcswitch.
+func (h *Htlcswitch) Validate() error {
+	if h.MailboxDeliveryTimeout < 0 {
```
Comment (positiveblue) [inline lncfg/htlcswitch.go:23]: q: shouldn't this be strictly positive?

**Review by positiveblue — APPROVED** [review]
LGTM :+1: 

My only concerns are about `mailboxdeliverytimeout` being 0 or too small (even when positive)

File: config.go (hunk 2 above)
Comment (yyforyongyu) [reply config.go:468]: yeah it applies to the whole function block

File: lncfg/htlcswitch.go (hunk 3 above)
Comment (yyforyongyu) [reply lncfg/htlcswitch.go:23]: good catch!

**Pull request merged**